### DISCUSSION TITLE
compiling the `wip/workplace_search` branch

### DISCUSSION THREAD

**Discussion by lukaszkuczynski** [dadoonet/fscrawler#1056] [Q&A] [answered]
hello, 
I am pretty interested in your work connected with the workplace search addon and already tried to compile it locally.
After downloading code from branch into my PC I have issues with compiling it. Because of the fact they say JDK14 is superseeded by JDK15 I am using this version of java and maven
```
mvn -version
Apache Maven 3.6.0
Maven home: /usr/share/maven
Java version: 15.0.1, vendor: Oracle Corporation, runtime: /usr/lib/jvm/jdk-15.0.1
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-54-generic", arch: "amd64", family: "unix"
```
When doing `mvn clean install` on this branch I got

```
[ERROR] Failed to execute goal org.sonatype.ossindex.maven:ossindex-maven-plugin:3.1.0:audit (audit-dependencies) on project fscrawler-workplacesearch-client: Detected 1 vulnerable components:
[ERROR]   org.eclipse.jetty:jetty-http:jar:9.4.34.v20201102:compile; https://ossindex.sonatype.org/component/pkg:maven/org.eclipse.jetty/jetty-http@9.4.34.v20201102?utm_source=ossindex-client&utm_medium=integration&utm_content=1.1.1
[ERROR]     * [CVE-2020-27218] In Eclipse Jetty version 9.4.0.RC0 to 9.4.34.v20201102, 10.0.0.alpha0 to 10.0.0.... (3.7); https://ossindex.sonatype.org/vuln/dbc80e9d-e821-4669-a4e1-6a537b6ad365?component-type=maven&component-name=org.eclipse.jetty.jetty-http&utm_source=ossindex-client&utm_medium=integration&utm_content=1.1.1
```

So far I am using bin that you compiled and placed on your dropbox. But it would be great to know how to compile this stuff.

**Comment by dadoonet** [comment, marked as answer]
Sounds like a recent CVE has been found. 
You can bypass this check for now. Have a look at https://fscrawler.readthedocs.io/en/latest/dev/build.html#check-for-vulnerabilities-cve

**Reply by lukaszkuczynski** [reply]
yes this command helped  `mvn clean install -Dossindex.fail=false`
thanks so much!

**Reply by lukaszkuczynski** [reply]
great job with the crawler, I was able to run 2.7 snapshot on my local files and indexed all the files like a charm. Maybe I will be able to help with some PR on that matter
